File: 0496-next-greater-element-i/solution.java
class Solution {
    public int[] nextGreaterElement(int[] nums1, int[] nums2) {
        int n1=nums1.length;int n2=nums2.length;
        List<Integer> l1=Arrays.stream(nums1).boxed().collect(Collectors.toList());
        List<Integer> l2=Arrays.stream(nums2).boxed().collect(Collectors.toList());
        List<Integer> res=new ArrayList<>();
        for(int i=0;i<n1;i++)
        {
            int index=l2.indexOf(nums1[i]);
            int max=nums2[index];
            for(int j=index;j<n2;j++)
            {
                if(nums2[j]>max)
                {
                    max=nums2[j];
                    break;
                }
            }
            if(nums1[i]==max)
                res.add(-1);
            else
                res.add(max);
        }
        int[] ans=res.stream().mapToInt(Integer::intValue).toArray();
        return ans;
    }
}

File: 0496-next-greater-element-i/solution.cpp
class Solution {
public:
    vector<int> nextGreaterElement(vector<int>& nums1, vector<int>& nums2) {
        stack<int> s;
        vector<int> res(nums1.size(),-1);
        unordered_map<int,int> m;
        for(auto ele:nums2)
        {
            while(!s.empty() && s.top()<ele)
            {
                m[s.top()]=ele;
                s.pop();
            }
            s.push(ele);
        }
        for(int i=0;i<nums1.size();i++)
        {
            if(m[nums1[i]])
                res[i]=m[nums1[i]];
        }
        return res;
        
    }
};
